src/pwm_sound_device.rs

```diff
@@ -0,0 +1,494 @@
+use crate::consts::*;
+use crate::util::*;
+use crate::timer::*;
+use alloc::vec::*;
+use core::mem::size_of;
+use core::ptr;
+use core::convert::TryFrom;
+use core::cmp::PartialEq;
+
+// macro_rules! ARM_DMACHAN_TI { ($( $chan:expr ),*) => (ARM_DMA_BASE + (($chan) * 0x100) + 0x08) }
+// macro_rules! ARM_DMACHAN_SOURCE_AD { ($( $chan:expr ),*) => (ARM_DMA_BASE + (($chan) * 0x100) + 0x0C) }
+// macro_rules! ARM_DMACHAN_DEST_AD { ($( $chan:expr ),*) => (ARM_DMA_BASE + (($chan) * 0x100) + 0x10) }
+// macro_rules! ARM_DMACHAN_TXFR_LEN { ($( $chan:expr ),*) => (ARM_DMA_BASE + (($chan) * 0x100) + 0x14) }
+// macro_rules! ARM_DMACHAN_STRIDE { ($( $chan:expr ),*) => (ARM_DMA_BASE + (($chan) * 0x100) + 0x18) }
+// macro_rules! ARM_DMACHAN_DEBUG { ($( $chan:expr ),*)	=>	(ARM_DMA_BASE + (($chan) * 0x100) + 0x20) }
+
+fn CleanAndInvalidateDataCacheRange(nAddress: usize, nLength: usize) {
+
+}
+
+fn ARM_DMACHAN_CONBLK_AD(chan: usize) -> usize  {
+    ARM_DMA_BASE + (chan * 0x100) + 0x04
+}
+fn BUS_ADDRESS(addr: usize) -> usize {
```

Comment:
This utility function can be moved into `const.rs` and have a more appropriate name.